modules/appeals_api/lib/appeals_api/hlr_pdf_constructor.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  class HlrPdfConstructor
+    PDF_TEMPLATE = Rails.root.join('modules', 'appeals_api', 'config', 'pdfs')
+
+    def initialize(higher_level_review_id)
+      @higher_level_review = AppealsApi::HigherLevelReview.find(higher_level_review_id)
+      veteran
+    end
+
+    def veteran
+      @veteran ||= build_veteran(@higher_level_review)
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    def build_veteran(higher_level_review)
+      form_data = higher_level_review.form_data['data']['attributes']
+      included = higher_level_review.form_data['included']
+      auth_headers = higher_level_review.auth_headers
+      OpenStruct.new(
+        first_name: auth_headers['X-VA-First-Name'],
+        middle_name: auth_headers['X-VA-Middle-Initial'],
+        last_name: auth_headers['X-VA-Last-Name'],
+        ssn: auth_headers['X-VA-SSN'],
+        va_file_number: auth_headers['X-VA-File-Number'],
+        service_number: auth_headers['X-VA-Service-Number'],
+        insurance_policy_number: auth_headers['X-VA-Insurance-Policy-Number'],
+        birth_date: auth_headers['X-VA-Birth-Date'],
+        address_line_1: form_data.dig('veteran', 'address', 'addressLine1'),
+        address_line_2: form_data.dig('veteran', 'address', 'addressLine2'),
+        city: form_data.dig('veteran', 'address', 'city'),
```

Comment:
I think we need to have a game plan for addressLine3

Reply:
there is no address 3 field on the PDF, the only 2 fields are these 
FieldType: Text
FieldName: F[0].#subform[2].CurrentMailingAddress_NumberAndStreet[0]
FieldNameAlt: 9. Current Mailing Address. Enter Number and Street.
FieldFlags: 25165824
FieldJustification: Left
FieldMaxLength: 30
---
FieldType: Text
FieldName: F[0].#subform[2].CurrentMailingAddress_ApartmentOrUnitNumber[0]
FieldNameAlt: 9. Enter Apartment or Unit Number.
FieldFlags: 25165824
FieldJustification: Left
FieldMaxLength: 5

We should document what those 2 fields expect

Reply:
As for zip-4, maybe we should add to the payload as an optional field since its on the form?

Reply:
yes zip 4 should always be optional or we run the risk of people needing to do a lookup to get the remaining 4 every time (and they might put down the wrong zip4 anyhow).

Reply:
I added it as optional. The dig will make sure it doesnt error on nil